src/core/dbus-manager.c

```diff
@@ -1409,16 +1409,37 @@ static int dump_impl(
 
         _cleanup_free_ char *dump = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        bool rate_limited= false;
         int r;
 
         assert(message);
 
-        /* Anyone can call this method */
-
+        /* 'status' access is the bare minimum always needed for this */
         r = mac_selinux_access_check(message, "status", error);
         if (r < 0)
                 return r;
 
+        /* Reload/reexec are also slow but non-destructive, use the same access check for dumps which, given
+         * the large amount of data to fetch, can stall for quite some time. If the caller is not privileged
+         * use rate limiting. */
+        r = mac_selinux_access_check(message, "reload", error);
```

Comment:
The reload check doesn't fail the call, it just enables rate limiting. But the caller must have "status" privileges at the very least. Otherwise it's a security policy compatibility change, and also doesn't make sense. We still want the security policy to be able to deny tout-court the action, not just cause it to be rate limited, and maintaining this check achieves that.